airflow/models/dagrun.py

```diff
@@ -1567,7 +1568,16 @@ def schedule_tis(
                         TI.run_id == self.run_id,
                         tuple_in_condition((TI.task_id, TI.map_index), schedulable_ti_ids_chunk),
                     )
-                    .values(state=TaskInstanceState.SCHEDULED)
+                    .values(
+                        state=TaskInstanceState.SCHEDULED,
+                        try_number=case(
+                            (
+                                or_(TI.state.is_(None), TI.state != TaskInstanceState.UP_FOR_RESCHEDULE),
```

Comment:
Same here

Reply:
what do you mean same here? please clarify what you think it should be

Reply:
oh i see i think you're saying same as https://github.com/apache/airflow/pull/39336#discussion_r1594675865 not same as the comment immediately above.  yeah this is needed because sql comparisons with null